### PR TITLE
Inquiry Artwork Brick

### PR DESCRIPTION
**Pull request by xtina-starr**
This creates a new inquiry artwork brick that has minimal metadata and includes the `isSelected` tap state.

new artwork brick

### CODE REVIEW

**Comment by yuki24** [comment]
Overall looks good to me :+1:

Also, this is just an idea, but the other thing we could do, instead of using `state`, is to use an actual checkbox in the component and [the :checked CSS selector](https://developer.mozilla.org/en-US/docs/Web/CSS/:checked). That way we don't have to use any state and the "what artworks did you purchase?" form will be able to capture the checkbox values directly. Also, anyone who is not familiar with React will be able to understand how the form works since this is just a plain html form. What do you think @xtina-starr @alloy @mzikherman @broskoski?

File: src/components/__stories__/inquiry_artwork.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { action, storiesOf } from "@kadira/storybook"
+import * as React from "react"
+import * as Relay from "react-relay"
+
+import Artwork from "../inquiry_artwork_brick/index"
```
Comment (alloy) [inline src/components/__stories__/inquiry_artwork.tsx]: FYI There’s no need to include the `index` filename, it is a special filename that by default is the entry point of the directory it resides in.

File: src/components/inquiry_artwork_brick/details.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
+import * as React from "react"
+import * as Relay from "react-relay"
+import styled from "styled-components"
+import colors from "../../assets/colors"
+import TextLink from "../text_link"
+
+export interface DetailsProps extends React.HTMLProps<ArtworkDetails> {
+  artwork: any
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/details.tsx]: You should extend from `RelayProps` as well and drop this `any` typing to get the actual types listed below https://github.com/artsy/reaction-force/pull/61/files#diff-bb6edb8ba51d5c558582b32f2884d355R88. E.g.

```ts
interface DetailsProps extends RelayProps, React.HTMLProps<ArtworkDetails> {}
```

(PS: I left out the `export` because I don’t see this interface being used outside of this file.)

File: src/components/inquiry_artwork_brick/details.tsx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+`
+
+export class ArtworkDetails extends React.Component<DetailsProps, null> {
+  artistLine() {
+    const { cultural_maker, artists } = this.props.artwork
+
+    if (cultural_maker) {
+      return <div><strong>{cultural_maker}</strong></div>
+    } else if (artists && artists.length) {
+      let artistLine = artists
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/details.tsx]: Use `const` instead of `let` if the variable is never going to change. I’m surprised the linter didn’t catch this 🤔

File: src/components/inquiry_artwork_brick/details.tsx (hunk 2 above)
Comment (alloy) [inline src/components/inquiry_artwork_brick/details.tsx]: Once you do this, you’ll notice that when you access `this.props.artwork` you’ll start getting completion for the fields listed in the `RelayProps` interface.

File: src/components/inquiry_artwork_brick/index.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  position: absolute;
+  z-index: 10;
+  top: 0;
+  left: 0;
+  opacity: 0.8;
+  text-align: center;
+  line-height: 350px;
+`
+
+export class ArtworkState {
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/index.tsx]: I don’t believe this is being used outside of this file, so `export` can be removed. (State details should probably _never_ be shared outside of a file, as they should be internal details.)

Additionally, I wonder if you really meant to use `class` here instead of `interface`? An interface is just a compile time thing that allows the compiler to check that the object conforms to a specific shape, whereas a class is an actual OOP class that can be instantiated, which we don’t do, as can be seen on line 53.

File: src/components/inquiry_artwork_brick/index.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+  isSelected: boolean
+}
+
+export class Artwork extends React.Component<RelayProps, ArtworkState> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      isSelected: false,
+    }
+  }
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/index.tsx]: Nit: missing newline

File: src/components/inquiry_artwork_brick/index.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+    let selectedOverlay
+    if (this.state.isSelected) {
+      selectedOverlay = (
+        <SelectedArtworkOverlay onClick={this.onSelect.bind(this)} > 
+          <Icon name="check" color="white" />
+        </SelectedArtworkOverlay>
+      )
+    }
+    return (
+      <div style={{position: "relative"}} >
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/index.tsx]: This should move to a styled component.

File: src/components/inquiry_artwork_brick/index.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    this.setState({
+      isSelected: !this.state.isSelected,
+    })
+  }
+
+  render() {
+    let selectedOverlay
+    if (this.state.isSelected) {
+      selectedOverlay = (
+        <SelectedArtworkOverlay onClick={this.onSelect.bind(this)} >
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/index.tsx]: I think you can use `pointer-events: none;` to make this element ignore events and thus they would be handled by the `ImageOuterContainer`. This only works with IE >= 11, what versions are we targeting?

File: src/components/inquiry_artwork_brick/index.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  }
+}
+
+export default Relay.createContainer(Artwork, {
+  fragments: {
+    artwork: () => Relay.QL`
+      fragment on Artwork {
+        image {
+          url(version: "large")
+          aspect_ratio
```
Comment (alloy) [inline src/components/inquiry_artwork_brick/index.tsx]: This field doesn’t appear to actually be used in the component.

**Comment by alloy** [comment]
Looking good thus far 👌

**Comment by alloy** [comment]
> the other thing we could do […] is to use an actual checkbox in the component and the :checked CSS selector. That way we don't have to use any state and the […] form will be able to capture the checkbox values directly.

While I like the idea of using HTML as closely as possible, I wonder if making assumptions about the implementation details of child components jives well with the React way. Because the child components _could_ change away from checkboxes to something else and the parent component wouldn’t know and effectively break. By using an API contract, especially a TypeScript one, any such change is hidden and would break the build.

File: src/components/inquiry_artwork_brick/details.tsx [hunk 9]
```diff
@@ -0,0 +1,102 @@
+import * as React from "react"
+import * as Relay from "react-relay"
+import styled from "styled-components"
+import colors from "../../assets/colors"
+import TextLink from "../text_link"
+
+export interface DetailsProps extends React.HTMLProps<ArtworkDetails> {
```
Comment (broskoski) [inline src/components/inquiry_artwork_brick/details.tsx]: As much as you can, try and extend the existing classes instead of duplicating them in a new folder. If any of that logic changes, we will now have to change it in two places.

File: src/components/inquiry_artwork_brick/details.tsx (hunk 2 above)
Comment (xtina-starr) [inline src/components/inquiry_artwork_brick/details.tsx]: So does that meet that the DetailsProps object should now be empty? It seems like I lose access to some of the props on artwork by doing that.

File: src/components/inquiry_artwork_brick/index.tsx (hunk 4 above)
Comment (xtina-starr) [inline src/components/inquiry_artwork_brick/index.tsx]: I don't quite understand this comment

File: src/components/inquiry_artwork_brick/details.tsx (hunk 2 above)
Comment (alloy) [inline src/components/inquiry_artwork_brick/details.tsx]: By lose, do you mean the TS compiler doesn’t allow you to access them? If so, which props?

File: src/components/inquiry_artwork_brick/index.tsx (hunk 4 above)
Comment (alloy) [inline src/components/inquiry_artwork_brick/index.tsx]: I assume the second paragraph? Let’s reboot it, why are you using `class` here instead of `interface`?

Comment (xtina-starr) [inline src/components/inquiry_artwork_brick/index.tsx]: Oh okay, yeah I should use interface here.

File: src/components/inquiry_artwork_brick/details.tsx (hunk 2 above)
Comment (xtina-starr) [inline src/components/inquiry_artwork_brick/details.tsx]: I was missing some of the properties on the artist object. I figured this out and realized I needed to add them to the `RelayProps` interface. However, now it doesn't like the types passed to the `reduce` method called on the artists collection.

**Comment by xtina-starr** [comment]
Does anyone have any more feedback for this PR?

File: src/components/inquiry_artwork_brick/index.tsx [hunk 10]
```diff
@@ -0,0 +1,115 @@
...
+import colors from "../../assets/colors"
+import Icon from "../icon"
+import ArtworkMetadata from "./metadata"
+
+const InquiryArtworkBrick = styled.div`
+  position: relative;
+`
+
+const ImageContainer = styled.div`
+  height: 350px;
```
Comment (l2succes) [inline src/components/inquiry_artwork_brick/index.tsx]: consider adding variable sizes based on the size of the screen (e.g. desktop vs mobile). We can do that in a separate PR when we add mobile support though.

For instance:

```
  height: ${props => props.size == Sizes.small ? 150 : 350 }
  ...
```

File: src/components/inquiry_artwork_brick/index.tsx [hunk 11]
```diff
@@ -0,0 +1,115 @@
...
+  display: table-cell;
+  vertical-align: middle;
+  width: 100%;
+  height: 100%;
+  text-align: center;
+`
+
+const Image = styled.img`
+  max-width: 100%;
+  max-height: 350px;
```
Comment (l2succes) [inline src/components/inquiry_artwork_brick/index.tsx]: make width and height variables defined at the top and use them here

```
max-height: ${defaultHeight};
max-width: ${defaultWidth};
```

File: src/components/inquiry_artwork_brick/index.tsx [hunk 12]
```diff
@@ -0,0 +1,115 @@
...
+  margin: auto;
+`
+
+const SelectedArtworkOverlay = styled.div`
+  background-color: ${colors.purpleRegular};
+  height: 350px;
```
Comment (l2succes) [inline src/components/inquiry_artwork_brick/index.tsx]: see above ^

File: src/components/inquiry_artwork_brick/index.tsx (hunk 11 above)
Comment (xtina-starr) [inline src/components/inquiry_artwork_brick/index.tsx]: just added this suggestion.

File: src/components/inquiry_artwork_brick/index.tsx (hunk 10 above)
Comment (xtina-starr) [inline src/components/inquiry_artwork_brick/index.tsx]: I agree that this can be added in a separate PR.

**Comment by xtina-starr** [comment]
@l2succes I made the change you recommended. Is this good to merge?

**Comment by l2succes** [comment]
Yup! Let's merge it.

**Comment by xtina-starr** [comment]
👍

**Comment by xtina-starr** [comment]
Do we check-in `dist/`??? @l2succes

**Comment by l2succes** [comment]
@xtina-starr Na we're not supposed to check in `dist` anymore. I added it to `.gitignore` in my PR so no worries, i'll take care of it.

**Comment by xtina-starr** [comment]
_phew!_
